### PR TITLE
Account for trips including the CapeFLYER

### PR DESCRIPTION
**Pull request by alicia-mbta**
#### Summary of changes
**Asana Ticket:** [Elixir.UndefinedFunctionError: function nil.mode/0 is undefined](https://app.asana.com/0/555089885850811/1200069554962417)

Fares are not specified for the CapeFLYER, so these changes make sure the trip planner logic doesn't break in this scenario and displays an additional message for more information instead.

image

### CODE REVIEW

**Comment by thecristen** [comment]
Looks super promising! Going to push it to Dev blue so we can review while being unable to geocode the destination (because of the Google API key thing)

**Review by thecristen — APPROVED** [review]
I'm hoping we get some design feedback to improve this further or content feedback to verify this is what we want to show... but code-wise it looks good!

File: apps/site/lib/site_web/views/trip_plan_view.ex [hunk 1]
```diff
@@ -663,4 +671,11 @@ defmodule SiteWeb.TripPlanView do
     |> Itinerary.transit_legs()
     |> Enum.any?(fn leg -> leg_is_from_or_to_airport?(leg) end)
   end
+
+  @spec itinerary_contains_capeflyer?(Itinerary.t()) :: boolean()
```
Comment (hannahpurcell) [inline apps/site/lib/site_web/views/trip_plan_view.ex:675]: Should both `itinerary_is_from_or_to_airport` and `itinerary_contains_capeflyer` be handled the same way?  Either as a function that takes itinerary, or as an assigned property in the render function

**Review by hannahpurcell — APPROVED** [review]
I learned about another mode of area transit :blush:  Looks good to me

File: apps/site/lib/site_web/views/trip_plan_view.ex (hunk 1 above)
Comment (alicia-mbta) [reply apps/site/lib/site_web/views/trip_plan_view.ex:675]: Good point -- I can probably refactor this.

**Comment by alicia-mbta** [comment]
Merging after getting confirmation from Yinka:

image

**Pull request merged**